### PR TITLE
margins: now in user units, and working for all aspect ratios cases

### PR DESCRIPTION
**Pull request by jonenst**
see #80

### CODE REVIEW

**Comment by jonenst** [comment]
Not sure how you test this, I used https://codepen.io/jonenst/pen/mdrgwoJ and copypaste to a local file and imported  the build like `<script src="file:///tmp/svg.panzoom.js/dist/svg.panzoom.js"></script>`

The code can probably become shorter by using more symmetries between meet and slice, but I wasn't sure it was worth it, I understand this one better.

File: src/svg.panzoom.js [hunk 1]
```diff
@@ -27,20 +27,143 @@ extend(Svg, {
...
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMID ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMID) {
+                viewportLeftOffset = widthOffset / 2
+                viewportRightOffset = -widthOffset / 2
+              } else if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMIN ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMIN) {
+                viewportLeftOffset = -widthOffset
```
Comment (Fuzzyma) [inline src/svg.panzoom.js]: Just looking at all the symmetry going on, this should be positive...

File: src/svg.panzoom.js [hunk 2]
```diff
@@ -27,20 +27,143 @@ extend(Svg, {
...
+              } else if (
+                preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMINYMAX ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMIDYMAX ||
+               preserveAspectRatio.align === preserveAspectRatio.SVG_PRESERVEASPECTRATIO_XMAXYMAX) {
+                viewportRightOffset = widthOffset
```
Comment (Fuzzyma) [inline src/svg.panzoom.js]: ...while this should be negative.

**Review by Fuzzyma — COMMENTED** [review]
Can you doublecheck that?

File: src/svg.panzoom.js (hunk 1 above)
Comment (Fuzzyma) [reply src/svg.panzoom.js]: Actually I think this should be `viewportRightOffset` and the other one `viewportLeftOffset`

**Comment by Fuzzyma** [comment]
oh no I hope the whole symmetry is not different now :D. I shorted the code by a lot using symmetry. we'll see ^^

**Comment by jonenst** [comment]
Yeah the slice mode was broken, the testing XMid/Min/Max and YMid/MinMax were swapped, and there was the bug you noticed, assigning to the wrong variable

**Comment by Fuzzyma** [comment]
@jonenst so I went ahead and simplified the code a bit. I would like you to glance over it in order to make sure that it is still correct. I pushed it to https://github.com/svgdotjs/svg.panzoom.js/pull/91
I would like to go break this down further but I don't want to break something on the line

**Pull request merged**